src/main/java/com/dj/core/model/graph/ConnectedNeuron.java

```diff
@@ -170,6 +170,24 @@ public void backwardSignalReceived(final Double error) {
 
         final var dzLearningRate = dz * context.getLearningRate();
         backwardConnections = backwardConnections.add(inputSignals.scalarMultiply(dzLearningRate));
+        if (context.getRegularizationRate() != 0.) {
+            backwardConnections.walkInColumnOrder(new RealMatrixChangingVisitor() {
```

Comment:
@b0noI 
backwardConnections is a 1:N vector.
So, in my view, it makes sense to use walkInRowOrder instead of walkInColumnOrder